pkg/core/proxy/integrations/mongo/scramAuth.go

```diff
@@ -315,6 +315,11 @@ func handleSaslStart(i int, actualMsg map[string]interface{}, expectedRequestSec
 	logger.Debug("fetch the conversationId for the SCRAM authentication", zap.String("cid", conversationID))
 	// generate the auth message from the recieved first request and recorded first response
 	authMessage := scram.GenerateAuthMessage(string(decodedActualReqPayload), newFirstAuthResponse, logger)
+	authMechanism, ok := actualMsg["mechanism"].(string)
```

Comment:
sasl start won't come in case of unauthorised, and if its authorised mechanism will be there always. So the empty case can be ignored.

Rest implementing the other two..incase the auth mechanism is something else it would ideally break for us on server side so failing here can help us add a new mechanism.